common/network-shuffle/src/main/java/org/apache/spark/network/shuffle/RemoteBlockPushResolver.java

```diff
@@ -568,6 +580,12 @@ public StreamCallbackWithID receiveBlockDataAsStream(PushBlockStream msg) {
     // getting killed. When this happens, we need to distinguish the duplicate blocks as they
     // arrive. More details on this is explained in later comments.
 
+    // Track if the block is received after shuffle merge finalize. The block would be considered
+    // as too late if it received after shuffle merge finalize, and hence mark it as a late block
+    // push to the pushMergeMetrics
+    if (partitionInfoBeforeCheck == null) {
+      pushMergeMetrics.lateBlockPushes.mark();
+    }
```

Comment:
Even if `partitionInfoBeforeCheck.mapTracker.contains(msg.mapIndex)`, it would be considered late.
Move `pushMergeMetrics.lateBlockPushes.mark();` to the `else` block below (before `new StreamCallbackWithID`) ?